src/repository.js

```diff
@@ -39,30 +40,89 @@ module.exports = class Repository {
     return entities
   }
 
+  /** 
+  *
+  * Find all method
+  * 
+  * @param {type}   object.limit Limit items to list  
+  * @param {type}   object.orderBy Order by query
+  *
+  * @return {type} List of entities
+  */
+  async findAll(options = {
```

Comment:
why do not expose `offset` as well?

Reply:
We will use `offset` to paginate data, right? So, I will implement a method called `findPaged` in another PR using` offset` and returning a model with pagination information. In this case, users can use the `find` method to obtain data using all parameters. If I include the `offset` parameter in` findAll`, this method doesn’t make sense, since it is the same as the` find` method, so I believe that the best way would be to just expose `find` in the base repository, what do you think ?

Reply:
It's the same situation as `findBy`, what do you think?

Reply:
For me `findAll` is just `find` with no `conditions` (search).

Thinking in terms of "job to be done", a developer could want to find all records but paged, even for "raw" pagination. 

And since it already have `limit` param, `offset` is something it could help as well, so I don't see why not pass it.

> It's the same situation as findBy, what do you think?

as commented bellow, I think we should remove `findBy` method

Reply:
Done!